tests/terraform/modules/join_rke2_master.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+# This script is used to join one or more nodes as masters to the first master
+set -x
+echo "$@"
+
+node_os=$1
+create_lb=$2
+initial_node_ip=$3
+token=$4
+rke2_version=$5
+public_ip=$6
+rke2_channel=$7
+server_flags=$8
+install_mode=$9
+rhel_username=${10}
+rhel_password=${11}
+install_method=${12}
+
+hostname=$(hostname -f)
+mkdir -p /etc/rancher/rke2
+cat <<EOF >>/etc/rancher/rke2/config.yaml
+write-kubeconfig-mode: "0644"
+tls-san:
+  - ${create_lb}
+server: https://${initial_node_ip}:9345
+token:  "${token}"
+node-name: "${hostname}"
```

Comment:
should we support more exhaustive config.yaml options?
ie 

cni: cilium

Reply:
This should cover that! It accepts a `.tfvars` file, and in there we have server and agent args, so passing something like:
`cni: cilium\nprofile: cis-1.6"` should work for both server and agent and add to this config.yaml. That should be present on line 33 from this file:
```
echo -e "$server_flags" >> /etc/rancher/rke2/config.yaml
```